proposals/000-cooperative-matrix-2.md

```diff
@@ -0,0 +1,536 @@
+
+- CoopMatTensorLayout<Dim, ClampMode>: Describes the layout of a tensor for addressing.
+  - Dim: The number of dimensions of the tensor layout (e.g., 2 for a 2D tensor).
+  - ClampMode: The clamping or boundary mode for tensor addressing (see CoopMatClampMode).
+  - Runtime states:
+    - blockSize[Dim]: Block size for each dimension.
+    - layoutDimension[Dim]: Logical size for each dimension.
+    - stride[Dim]: Stride (in elements) for each dimension.
+    - offset[Dim]: Offset for each dimension.
+    - span[Dim]: Span for each dimension.
+
+#### CoopMatTensorView
+
+Similarly, CoopMatTensorView can be thought of as somewhat similar to a "TextureView" in texture handling. It allows for reinterpreting or accessing a given cooperative matrix tensor in a specific way without modifying the underlying data.
+
+- CoopMatTensorView<Dim, HasDimensions, p0, p1 ... p{Dim-1} >: Describes a view into a tensor.
+  - Dim: The number of dimensions of the tensor view.
+  - HasDimensions: Boolean indicating whether explicit dimensions are set for the view.
+  - p0, p1, ... p{Dim-1}: A compile-time list specifying the permutation of dimensions for the view. Each pi indicates which source dimension is mapped to the i-th dimension of the view.
+  - Runtime states:
+    - permutation[Dim]: Permutation of dimensions (compile-time constant).
+    - viewDimension[Dim]: Logical size for each dimension in the view when HasDimensions is true.
+    - viewStride[Dim]: Stride (in elements) for each dimension in the view.
+    - clipRowOffset: Offset for the row clipping region.
+    - clipRowSpan: Span for the row clipping region.
+    - clipColOffset: Offset for the column clipping region.
+    - clipColSpan: Span for the column clipping region.
+
+CoopMatTensorView takes (N + 2) number of generic arguments where N is a value given as a first generic argument. As an example, when the first value is 2, CoopMatTensorView must use 4 generic arguments. More examples will be `CoopMatTensorView<1, false, 0>`, `CoopMatTensorView<2, true, 0, 1>` and `CoopMatTensorView<3, false, 0, 2, 1>`.
+
+The first generic argument to `CoopMatTensorView` indicates the dimensionality, and the trailing values indicates the re-ordering of the dimension. A common use cases are `CoopMatTensorView<2, false, 0, 1>` for row-major-matrix load and store, and `CoopMatTensorView<2, false, 1, 0>` for column-major-matrix load and store.
+
+Note that CoopMatTensorLayout and CoopMatTensorView are only for loading and storing of CoopMat instances.
+
+### New Enums
+
+- CoopMatScope: Specifies the hardware scope over which the cooperative matrix is distributed.
+  - Device: The entire device (all invocations on the device participate).
+  - Workgroup: The current workgroup (all invocations in the workgroup participate).
```

Comment:
You should say which scopes are supported, and for Workgroup how does a developer know whether Workgroup is supported (requires a Vulkan coopmat2 feature to be enabled)?

Similarly, how does a developer know whether all the other coopmat2 features are supported? Will slang emulate any of them on coopmat1 implementations?